extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -170,33 +190,83 @@ public void emit(final Event event)
       Map<String, Object> result = resultBuilder.build();
 
       try {
-        String resultJson = jsonMapper.writeValueAsString(result);
-        ObjectContainer<String> objectContainer = new ObjectContainer<>(
-            resultJson,
-            StringUtils.toUtf8(resultJson).length
+        byte[] resultBytes = jsonMapper.writeValueAsBytes(result);
+        ObjectContainer<byte[]> objectContainer = new ObjectContainer<>(
+            resultBytes,
+            resultBytes.length
         );
         if (event instanceof ServiceMetricEvent) {
-          if (!metricQueue.offer(objectContainer)) {
+          if (config.getMetricTopic() == null || !metricQueue.offer(objectContainer)) {
             metricLost.incrementAndGet();
           }
         } else if (event instanceof AlertEvent) {
-          if (!alertQueue.offer(objectContainer)) {
+          if (config.getAlertTopic() == null || !alertQueue.offer(objectContainer)) {
             alertLost.incrementAndGet();
           }
         } else if (event instanceof RequestLogEvent) {
           if (config.getRequestTopic() == null || !requestQueue.offer(objectContainer)) {
             requestLost.incrementAndGet();
           }
+        } else if (event instanceof SegmentMetadataEvent) {
+          if (config.getSegmentMetadataTopic() == null ) {
+            segmentMetadataLost.incrementAndGet();
+          } else {
+            switch (config.getSegmentMetadataTopicFormat()) {
+              case protobuf:
+                resultBytes = convertMetadataEventToProto((SegmentMetadataEvent) event, segmentMetadataLost);
+                objectContainer = new ObjectContainer<>(
+                    resultBytes,
+                    resultBytes.length
+                );
+                break;
+              case json:
+                // Do Nothing. We already have the JSON object stored in objectContainer
+                break;
+              default:
+                throw new InvalidConfigurationException("segment.metadata.topic.format has an invalid value " + config.getSegmentMetadataTopicFormat().toString());
```

Comment:
we should catch invalid configuration at configuration parsing time so that it doesn't become possible to get into this state. We should still throw an exception, in case we get to this point and we have an invalid format, but then we would usually throw an UnsupportedOperationException (druid typically uses its own UOE class for this) indicating that the code got into a state it shouldn't have.

It also doesn't make sense to use InvalidConfigurationException in this case, since this class is part of the Kafka APIs and this code is not related to Kafka configurations.